core/src/main/java/hudson/console/HyperlinkNote.java

```diff
@@ -69,7 +70,7 @@ public ConsoleAnnotator annotate(Object context, MarkupText text, int charPos) {
                 url = Jenkins.get().getRootUrl()+url.substring(1);
             }
         }
-        text.addMarkup(charPos, charPos + length, "<a href='" + url + "'"+extraAttributes()+">", "</a>");
+        text.addMarkup(charPos, charPos + length, "<a href='" + Util.escape(url) + "'"+extraAttributes()+">", "</a>");
```

Comment:
Note that this code is using `'` where most generated markup uses `"`, though it should not make any difference (`Util.escape` will escape both if they happen to be present in the URL).